### PR TITLE
Enhance test for managed and external delta table location validation

### PR DESCRIPTION
**Pull request by krvikash**
## Description
This PR improves the `testCorruptedTableLocation` to verify for managed and external delta table locations

Follow up of https://github.com/trinodb/trino/pull/16651#discussion_r1147377727

## Release notes

(X) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 1]
```diff
@@ -115,55 +117,82 @@ public void testNoColumnStats()
...
+        assertQueryReturnsEmptyResult("SELECT column_name, data_type FROM system.jdbc.columns WHERE table_cat = CURRENT_CATALOG AND table_schem = CURRENT_SCHEMA AND table_name LIKE 'bad\\_person\\_%' ESCAPE '\\'");
 
         // DROP TABLE should succeed so that users can remove their corrupted table
         getQueryRunner().execute("DROP TABLE " + tableName);
         assertFalse(getQueryRunner().tableExists(getSession(), tableName));
+        if (isManaged) {
+            assertFalse(tableLocation.toFile().exists(), "Table location should not exist");
+        }
+        else {
+            assertTrue(tableLocation.toFile().exists(), "Table location should exist");
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:194]: nit: I would recommend using `assertThat(tableLocation.toFile())...`. It shows the directory status when failed.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 2]
```diff
@@ -115,55 +117,82 @@ public void testNoColumnStats()
...
     @Test
-    public void testCorruptedTableLocation()
+    public void testCorruptedManagedTableLocation()
+            throws Exception
+    {
+        String tableName = "bad_person_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 id, 'person1' name", 1);
+        String tableLocation = (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*$', '') FROM " + tableName);
+        // Remove leading 'file:' because $path column returns 'file:/path-to-file' in case of local file system
+        testCorruptedTableLocation(tableName, Path.of(tableLocation.replaceFirst("^file:", "")), true);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:127]: nit: We could use `Path.of(URI.create(tableLocation))` instead.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 3]
```diff
@@ -115,55 +117,82 @@ public void testNoColumnStats()
...
 
-        // break the table by deleting all its files including transaction log
-        deleteRecursively(tableLocation, ALLOW_INSECURE);
+    private void testCorruptedTableLocation(String tableName, Path tableLocation, boolean isManaged)
+            throws Exception
+    {
+        Path transactionLogDirectory = tableLocation.resolve("_delta_log");
+
+        // break the table by deleting all its files under transaction log
+        deleteRecursively(transactionLogDirectory, ALLOW_INSECURE);
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:149]: Previously, the test deleted the whole table location. Could you leave the motivation to the commit body?

**Review by ebyhr — APPROVED** [review]

**Comment by krvikash** [comment]
Thanks, @ebyhr for the review. Addressed comments.

**Pull request merged**